recipes/dev/generate_v2.py

```diff
@@ -0,0 +1,193 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import itertools
+import sys
+import time
+from typing import Any, Dict, List
+
+import torch
+from omegaconf import DictConfig
+
+from torchtune import config, training, utils
+from torchtune.data import (
+    left_pad_sequence,
+    load_image,
+    Message,
+    padded_collate_tiled_images_and_mask,
+)
+
+from torchtune.generation import sample
+
+from torchtune.modules.transforms import Transform
+
+
+class SingleTurnYAMLToMessages(Transform):
+    """
+    Converts a single turn conversation in YAML format to a list of messages.
+    """
```

Comment:
Just curious, why does it have to be in YAML format? Like why not just call .to_dict or whatever so this isn't as tied to configs? 

Separately, is this the long-term home for this class or should it go in data?

Reply:
Recipes ARE tied to configs though. So, this recipe needs a way to translate YAML to Message. 

The core of the recipe can be ripped out / modified in any way now that things are in messages. Eventually, this should go in data, but not until we get some usage.

Reply:
By YAML you mean an omegaconf DictConfig, right? That's why I'm suggesting to just use .to_dict cause then we get a primitive type straightaway (I think). But anyways I don't disagree with your points here

Reply:
I'm being dumb but I'm not sure I understand. Can you show a code sample?

Reply:
I mean in practice it's basically the same thing since the signature in `SingleTurnYAMLToMessages` already takes `Dict[str, Any]`. But really you are doing `self.to_messages(cfg.prompt)` and (I think) `cfg.prompt` is an `OmegaConf.DictConfig`. So might be better to just do `self.to_messages(OmegaConf.to_container(cfg.prompt))` so that `SingleTurnYAMLToMessages` is actually just taking a vanilla dict rather than some OmegaConf class.